### PR TITLE
[Enhancement] Automatically loading custom dependencies when inject custom component factory

### PR DESCRIPTION
**Pull request by heshan0131**
- Automatically provide custom factory dependencies while injecting them
- Create an example demo using custom components

screen shot 2019-03-06 at 1 14 21 pm
[__add a custom sidebar that scales down the side panel, and move it to the edge of the screen__]

### CODE REVIEW

**Comment by heshan0131** [comment]
https://github.com/uber/kepler.gl/issues/356

File: src/components/container.js [hunk 1]
```diff
@@ -167,6 +167,13 @@ export function injectComponents(recipes) {
         Console.error(errorMsg.wrongPairType);
         return inj;
       }
+
+      // collect dependencies of custom factories, if there is any.
+      // Add them to the injector
+      const customDependencies = flattenDeps([], recipe[1]);
```
Comment (heshan0131) [inline src/components/container.js]: @btford

File: examples/replace-component/src/actions.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import {createAction} from 'redux-actions';
+
+export const setMapConfig = createAction(
+  'SET_MAP_CONFIG',
+  payload => payload
```
Comment (btford) [inline examples/replace-component/src/actions.js]: Does `createAction` require a second arg if the payload formatter is the identity fn?

File: src/components/container.js (hunk 1 above)
Comment (btford) [reply src/components/container.js]: 👍

File: examples/replace-component/src/actions.js (hunk 2 above)
Comment (heshan0131) [reply examples/replace-component/src/actions.js]: certain version of redux-actions will throw error if payload creator is undefined.

**Pull request merged**